gpt4all-bindings/typescript/prompt.cc

```diff
@@ -2,145 +2,196 @@
-                return isRecalculating;
-            },
-            wrapper->promptContext);
-
-        // Update the C context by giving access to the wrappers raw pointers to std::vector data
-        // which involves no copies
-        ctx->logits = wrapper->promptContext.logits.data();
-        ctx->logits_size = wrapper->promptContext.logits.size();
-        ctx->tokens = wrapper->promptContext.tokens.data();
-        ctx->tokens_size = wrapper->promptContext.tokens.size();
-
-        // Update the rest of the C prompt context
-        ctx->n_past = wrapper->promptContext.n_past;
-        ctx->n_ctx = wrapper->promptContext.n_ctx;
-        ctx->n_predict = wrapper->promptContext.n_predict;
-        ctx->top_k = wrapper->promptContext.top_k;
-        ctx->top_p = wrapper->promptContext.top_p;
-        ctx->temp = wrapper->promptContext.temp;
-        ctx->n_batch = wrapper->promptContext.n_batch;
-        ctx->repeat_penalty = wrapper->promptContext.repeat_penalty;
-        ctx->repeat_last_n = wrapper->promptContext.repeat_last_n;
-        ctx->context_erase = wrapper->promptContext.contextErase;
-
-        _config.mutex->unlock();
+        _promptCallbackFn =
+            Napi::ThreadSafeFunction::New(config.promptCallback.Env(), config.promptCallback, "PromptWorker", 0, 1, this);
     }
+}
 
-    void PromptWorker::OnOK()
+PromptWorker::~PromptWorker()
+{
+    if (_config.hasResponseCallback)
     {
-        promise.Resolve(Napi::String::New(Env(), result));
+        _responseCallbackFn.Release();
     }
-
-    void PromptWorker::OnError(const Napi::Error &e)
+    if (_config.hasPromptCallback)
     {
-        promise.Reject(e.Value());
+        _promptCallbackFn.Release();
     }
-
-    Napi::Promise PromptWorker::GetPromise()
+}
+
+void PromptWorker::Execute()
+{
+    _config.mutex->lock();
+
+    LLModelWrapper *wrapper = reinterpret_cast<LLModelWrapper *>(_config.model);
+
+    auto ctx = &_config.context;
+
+    if (size_t(ctx->n_past) < wrapper->promptContext.tokens.size())
+        wrapper->promptContext.tokens.resize(ctx->n_past);
+
+    // Copy the C prompt context
+    wrapper->promptContext.n_past = ctx->n_past;
+    wrapper->promptContext.n_ctx = ctx->n_ctx;
+    wrapper->promptContext.n_predict = ctx->n_predict;
+    wrapper->promptContext.top_k = ctx->top_k;
+    wrapper->promptContext.top_p = ctx->top_p;
+    wrapper->promptContext.temp = ctx->temp;
+    wrapper->promptContext.n_batch = ctx->n_batch;
+    wrapper->promptContext.repeat_penalty = ctx->repeat_penalty;
+    wrapper->promptContext.repeat_last_n = ctx->repeat_last_n;
+    wrapper->promptContext.contextErase = ctx->context_erase;
+
+    // Napi::Error::Fatal(
+    //         "SUPRA",
+    //         "About to prompt");
+    // Call the C++ prompt method
+
+    wrapper->llModel->prompt(
+        _config.prompt,
+        _config.promptTemplate,
+        [this](int32_t token_id) { return PromptCallback(token_id); },
+        // [](int32_t tid) { return true; },
```

Comment:
This commented out code can probably be removed.

Reply:
removed